### PR TITLE
Move to capturePage over display media for screenshot handling

### PR DESCRIPTION
**Pull request by Tyriar**

### CODE REVIEW

**Review by meganrogge — DISMISSED** [review]

**Comment by bpasero** [comment]
I think this method needs to be put into `IHostService` and then implemented in native + web if possible.

**Comment by Tyriar** [comment]
@bpasero perfect thanks! Was probably going to reach out to ask what the right way to deal with that was

**Comment by bpasero** [comment]
I think we do have cases where in web we simply throw from the host service impl because we know its not going to be used from web, but thats a shortcut if you need to move fast and there is no straight forward web impl.

File: src/vs/platform/native/electron-main/nativeHostMainService.ts [hunk 1]
```diff
@@ -687,6 +687,17 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 	//#endregion
 
 
+	//#region Screenshots
+
+	async getScreenshot(windowId: number | undefined): Promise<ArrayBufferLike | undefined> {
+		const window = this.windowById(undefined, windowId);
```
Comment (bpasero) [inline src/vs/platform/native/electron-main/nativeHostMainService.ts:693]: Do we need to account for auxiliary windows? In the end, chat might be in an aux window and then using the primary window `webContents` could be wrong?

**Review by meganrogge — APPROVED** [review]
Thanks!

File: src/vs/platform/native/electron-main/nativeHostMainService.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/platform/native/electron-main/nativeHostMainService.ts:693]: I'll check how it's handled there, I guess almost always at least in this initial use case we want the screenshot of the primary window.

Comment (Tyriar) [reply src/vs/platform/native/electron-main/nativeHostMainService.ts:693]: Handling it the same as most users of `windowById`: https://github.com/microsoft/vscode/pull/231410

**Pull request merged**